src/bootstrap/builder.rs

```diff
@@ -787,6 +787,27 @@ impl<'a> Builder<'a> {
             }
         }
 
+        // If we're statically linking in an LLVM compiled for ThinLTO,
+        // configure the linker accordingly.
+        if Mode::Rustc == mode && self.config.llvm_thin_lto && !self.config.llvm_link_shared {
+            if target.contains("msvc") {
+                // Here we assume the linker is lld-link.exe.
+                rustflags.arg(&format!("-Clink-arg=/opt:lldtojobs={}", self.jobs()));
+            } else {
+                // Here we assume the linker is clang. If not, there'll be
+                // linker errors.
+                rustflags.arg("-Clink-arg=-fuse-ld=lld");
```

Comment:
It's also worth pointing out that you're assuming that `lld` is available otherwise this will fail even if the linker is `clang`.

Reply:
done